server/src/test/java/org/elasticsearch/cluster/routing/allocation/shards/ShardsAvailabilityHealthIndicatorServiceTests.java

```diff
@@ -922,17 +983,20 @@ public void testRestoreFromSnapshotReportsFeatureStates() {
             List.of(),
             List.of()
         );
-
         // add the data stream to the cluster state
+        ProjectMetadata projectMetadata = clusterState.metadata().getProject(projectId);
         Metadata.Builder mdBuilder = Metadata.builder(clusterState.metadata())
-            .put(newInstance(featureDataStreamName, List.of(backingIndex.getIndex())));
+            .put(
+                ProjectMetadata.builder(projectMetadata).put(newInstance(featureDataStreamName, List.of(backingIndex.getIndex()))).build()
+            );
         ClusterState state = ClusterState.builder(clusterState).metadata(mdBuilder).build();
```

Comment:
Nit: you can do
```java
var projectBuilder = ProjectMetadata.builder(clusterState.metadata().getProject(projectId))
	.put(newInstance(featureDataStreamName, List.of(backingIndex.getIndex())));
ClusterState state = ClusterState.builder(clusterState).putProjectMetadata(projectBuilder).build();
```
that makes things a bit more compact. That applies to several other places in this class as well.